### PR TITLE
Add pds logger

### PR DESCRIPTION
**Pull request by juzen2003**
- Based on https://github.com/SETI/pds-hst-pipeline/pull/21
- Add pds logger (replace logging)
- A directory: ```logs``` will be created automatically in TMP_WORKING_DIR, and logs of pipeline for each proposal id will be stored there with this format: ```f"hst_{proposal_id}_pipeline_log_{date_time}.log"```. 
- A file ```ERRORS.log``` will also be created, error messages will be logged there. New error messages will be appended.

### CODE REVIEW

**Review by rfrenchseti — COMMENTED** [review]
Some overall comments not necessarily related to these changes:

- Just taking a quick look at Dave's logging code, I'm wondering if there isn't another problem that should be fixed. The code is full of assert statements. I would think those are inappropriate for a piece of software that's going to run in an automated pipeline. Instead the conditions should be checked for and if there is a problem a log entry should be made instead, possibly with the pipeline aborting at that point as well.
- Also we currently use mixed-case filenames (InsertChanges.py). This violates PEP8, which says "Modules should have short, all-lowercase names. Underscores can be used in the module name if it improves readability. Python packages should also have short, all-lowercase names, although the use of underscores is discouraged."

**Comment by juzen2003** [comment]
- Replace all assert statement with raise exception.
- The change of file names is in this pull: https://github.com/SETI/pds-hst-pipeline/pull/24

File: Pipeline.py [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
+        limits={
+            "info": INFO_MESSAGE_LIMIT,
+        },
+    )
     state_machine = StateMachine(dirs, proposal_id)
     state_machine.run()
+    PDS_LOGGER.close()
 
 
 if __name__ == "__main__":
```
Comment (markshowalter) [inline Pipeline.py:27]: This looks really good. When you open a logger, as with opening a file, you should take care to be 100% certain the logger gets closed, even in the event of an exception. So I recommend you put "try:" immediately after PDS_LOGGER.open, and put "finally:" before PDS_LOGGER.close(). Also, PdsLogger has a handy method to put an exception into the log file. Before the "finally:", add `except Exception as e: PDS_LOGGER.error(e)`. You will get the error message and stacktrace in the log file, so you can see exactly what went wrong.

File: pdart/documents/Downloads.py [hunk 2]
```diff
@@ -33,12 +35,13 @@ def download_product_documents(proposal_id: int, download_dir: str) -> Set[str]:
...
-
     res: Set[str] = set()
-
+    PDS_LOGGER.open("Download product documents")
     for (url, basename) in table:
         filepath = fs.path.join(download_dir, basename)
         if _retrieve_doc(url, filepath):
+            PDS_LOGGER.log("info", f"Retrieve {basename} from {url}")
             res.add(basename)
+    PDS_LOGGER.close()
```
Comment (markshowalter) [inline pdart/documents/Downloads.py:45]: Good. Add another "try"/"finally" here.

**Review by markshowalter — APPROVED** [review]
Nice work! This all looks great. Aside from my comments about try/finally, this is good to go. Later, I would like to see some example log files, and maybe we can review the events that get logged. For example, it would make sense to add logging to the target identification code I gave you long ago, telling us what target was identified and maybe warning us if the target identification was uncertain. However, we can address that after you merge this set of changes.

**Comment by juzen2003** [comment]
- Add try/finally to make sure PDS_LOGGER is closed when exception happened.
- Update requirements.txt to work with Python 3.9.10

**Pull request merged**